### PR TITLE
Add `direction_matrix` kwarg to `ImageData.__init__`

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Follow-up to https://github.com/pyvista/pyvista/pull/6612. `direction_matrix` is used alongside `spacing` and `origin` to define the ImageData orientation and so I think it makes sense to include this in `__init__`.

### CODE REVIEW

**Comment by user27182** [comment]
This should be good to go but should wait until #6726 is merged first

File: pyvista/core/grid.py
Comment (user27182) [inline pyvista/core/grid.py:587]: This is technically a breaking change, might be best to put the new keyword at the end.
```suggestion
        deep=False,
        direction_matrix=None,
```

Comment (tkoyama010) [reply pyvista/core/grid.py:587]: +1 not breaking change.

File: pyvista/core/grid.py
Comment (tkoyama010) [inline pyvista/core/grid.py:587]: We should add explanations for each parameter in the docstring :)

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: pyvista/core/grid.py
Comment (user27182) [reply pyvista/core/grid.py:587]: whoops! 😅 

Odd that there are no complaints from pre-commit about this (numpydoc or ruff should be linting this kind of thing...)

Comment (tkoyama010) [reply pyvista/core/grid.py:587]: ref #6755

Comment (user27182) [reply pyvista/core/grid.py:587]: > Let's verify if #6755 can catch this error after merging it.

Still not caught by pre-commit, unfortunately :/

Comment (user27182) [reply pyvista/core/grid.py:587]: This might be the issue. This line should be removed:
https://github.com/pyvista/pyvista/blob/ea732e95990b1f3ad56ca1ee3e514142f6d66b59/pyproject.toml#L309

Comment (user27182) [reply pyvista/core/grid.py:587]: Hmm... this isn't it either. That doesn't check for missing members. Plus we define init params in the class docstring, not the __init__ docstring.

**Review by banesullivan — DISMISSED** [review]

**Comment by banesullivan** [comment]
@user27182, would you please handle the missing type annotation here and update this branch?

**Review by tkoyama010 — APPROVED** [review]
LGTM

File: pyvista/core/grid.py
Comment (tkoyama010) [inline pyvista/core/grid.py:16]:
```suggestion
if TYPE_CHECKING:  # pragma: no cover
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**
